dspace/config/modules/discovery.cfg

```diff
@@ -22,3 +22,10 @@ discovery.index.projection=dc.title,dc.contributor.*,dc.date.issued
 # 1) you need to set the DiscoverySearchRequestProcessor in the dspace.cfg 
 # 2) to show facet on Site/Community/etc. you need to add a Site/Community/Collection
 #	 Processors plugin in the dspace.cfg
+
+# Allow auto-reindexing
+# When enabled, if any database migrations are applied to your database (via Flyway), then a reindex flag 
+# is written to [dspace]/solr/search/conf/reindex.flag. Whenever the DSpace webapp is (re)started, it checks
+# for the existence of that file. If found, a background reindex of all content is triggered in Discovery.
+# Defaults to true: auto-reindexing is enabled.
```

Comment:
I don't think that the above description matches the actual implementation. If the reindex file flag is present the reindex thread should start regardless to the value of the discovery.autoReindex configuration.

The configuration should only be checked to decide if the file flag need to be created or not.

I recommend to implement the feature as described above instead than fix the comment because in this way we could  create the flag file in an external procedure if needed and have it respected

Reply:
To clarify here, this PR is a forward port of #2184 (which was merge/applied to 6.x).  Therefore, if we want to tweak the text here, that's fine by me.  But, I'd recommend we not require a new implementation, as that'd turn this into a separate feature.  In my opinion, a new implementation should be a separate JIRA ticket.

Reply:
I have created the ticket https://jira.duraspace.org/browse/DS-4247 to track the suggestion for a future improvement. 
Can we update the comment as follow?

```
# If any database migrations are applied to your database (via Flyway), then a reindex flag 
# is always written to [dspace]/solr/search/conf/reindex.flag. 
# Whenever the DSpace webapp is (re)started, it checks
# if the auto-reindexing proprerty is enabled AND a such file exists. If the two conditions are 
# satisfied, a background reindex of all content is triggered in Discovery.
# Defaults to true: auto-reindexing is enabled.
```

Reply:
Comment updated.  Thank you for discovering this omission.